model/lifecycle.go

```diff
@@ -1442,14 +1442,28 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
 		for _, taskname := range pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if _, ok := existingTasksIndex[taskname]; ok {
+			if t, ok := existingTasksIndex[taskname]; ok {
+				if !t.Activated && activated {
+					// update task activation
+					if _, err = t.ActivateTask(evergreen.User); err != nil {
```

Comment:
I deliberated between ActivateTask and SetActiveState, but it seems like SetActiveState does too much.

Reply:
I think `ActivateTask` will not update the build cache, though.

Reply:
Doesn't look like SetActiveState does either. I'll pull RefreshTaskCache out so we still do that even if we don't create tasks.